aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -844,6 +845,25 @@ public String getUsername() {
         }
     }
 
+    /**
+     * Returns the Cognito User Sub attribute of the current access token.
+     * @return The sub attribute of the current access token.
+     * @see <a href="https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#amazon-cognito-user-pools-using-the-access-token">Using the Access Token</a>
+     * from Cognito documentation.
+     */
+    @AnyThread
+    public String getUserSub() {
+        try {
+            if (userpoolsLoginKey.equals(mStore.get(PROVIDER_KEY))) {
+                String token = mStore.get("token");
+                return (String) CognitoJWTParser.getPayload(token).get("sub");
```

Comment:
is this equivalent to the following?
```
return CognitoJWTParser.getClaim(token, "sub");
```

Reply:
I think so? I'm not 100% sure.

Reply:
they do the [exactly same thing](https://github.com/aws-amplify/aws-sdk-android/blob/1940e0500f50e34b5096e4635a4b60beb0fa8284/aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/util/CognitoJWTParser.java#L105-L106) :) I was only suggesting it because it's much cleaner and this is exactly what it's supposed to be used for